app/controllers/participants/validations_controller.rb

```diff
@@ -0,0 +1,237 @@
+
+    def cannot_find_details
+      if request.put?
+        if step_valid?
+          store_validation_data!
+          reset_form_data
+          redirect_to_step :complete
+        end
+      end
+    end
+
+    def complete
+      @school = participant.school
+      if participant.ecf_participant_eligibility.present? && participant.ecf_participant_eligibility.eligible_status?
+        # TRN has been validated, qts, no flags, not done before, no previous induction
+        render_completed_page
+      else
+        # all other cases
+        render_manual_check_page
+      end
+    end
+
+    def reset
+      reset_form_data
+      redirect_to participants_validation_start_path
+    end
+
+  private
+
+    def render_completed_page
+      if participant.school_cohort.full_induction_programme?
+        @partnership = @school.partnerships.active.find_by(cohort: participant.school_cohort.cohort)
+        render "complete_fip"
+      else
+        render "complete_cip"
+      end
+    end
+
+    def render_manual_check_page
+      if participant.school_cohort.full_induction_programme?
+        @partnership = @school.partnerships.active.find_by(cohort: participant.school_cohort.cohort)
+        render "manual_details_check_fip"
+      else
+        render "manual_details_check_cip"
+      end
+    end
+
+    def check_not_already_completed
+      redirect_to_step :complete if complete?
+    end
+
+    def complete?
+      participant.ecf_participant_validation_data.present? || participant.ecf_participant_eligibility.present?
+    end
+
+    def validate_participant_details_and_redirect
+      result = ParticipantValidationService.validate(trn: @form.trn,
+                                                     full_name: @form.name,
+                                                     date_of_birth: @form.date_of_birth,
+                                                     nino: @form.national_insurance_number)
+      if result.nil?
+        @form.increment_validation_attempts
+        redirect_to_step :cannot_find_details
+      else
+        store_trn!(result[:trn])
+        eligibility_data = store_eligibility_data!(result)
+        # if not eligibile store validation data for re-check later
+        store_validation_data! unless eligibility_data.eligible_status?
+        reset_form_data
+        redirect_to_step :complete
+      end
+    rescue StandardError => e
+      Rails.logger.error("Problem with DQT API: " + e.message)
+      store_validation_data!(api_failure: true)
+      reset_form_data
+      redirect_to_step :complete
+    end
+
+    def store_validation_data!(opts = {})
+      participant.create_ecf_participant_validation_data!({
+        trn: @form.trn,
+        full_name: @form.name,
+        date_of_birth: @form.date_of_birth,
+        nino: @form.national_insurance_number,
+      }.merge(opts))
+    end
+
+    def store_trn!(trn)
+      if participant.teacher_profile.trn.present? && participant.teacher_profile.trn != trn
```

Comment:
This should probably fall to manual validation

Reply:
Yes, wasn't sure how we wanted to handle this